internal/pkg/term/selector/ec2.go

```diff
@@ -67,27 +67,44 @@ func (s *EC2Select) VPC(prompt, help string) (string, error) {
 
 // PublicSubnets has the user multiselect public subnets given the VPC ID.
 func (s *EC2Select) PublicSubnets(prompt, help, vpcID string) ([]string, error) {
-	return s.subnet(prompt, help, vpcID, ec2.FilterForPublicSubnets())
+	return s.subnet(prompt, help, vpcID, true)
```

Comment:
boolean params are usually a codesmell because it's not clear what the boolean stands for: https://martinfowler.com/bliki/FlagArgument.html

```go
return s.selectPublicSubnets(prompt, help, vpcID)
return s.selectPrivateSubnets(prompt, help, vpcID)

func (s *EC2Select) selectPublicSubnets(prompt, help, vpcID string) ([]string, error) {
   subnets, err := s.ec2.ListVPCSubnets(vpcID)
   if err != nil { ... }
   publicSubnets := .... // for-loop and filter on .Public
   
   return s.selectSubnets(prompt, help, publicSubnets) // This method is common for s.selectPrivateSubnets
}
```